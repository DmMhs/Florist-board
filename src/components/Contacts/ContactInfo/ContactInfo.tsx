import React, { useContext } from 'react';

import './ContactInfo.less';
import labels from '../../../config/labels';
import { AuthContext } from '../../Auth/AuthContext';

interface ContactInfoProps {
  address: string;
  phone: string;
  email?: string;
  telegram?: string;
  facebook?: string;
  instagram?: string;
}

const contactInfo = (props: ContactInfoProps) => {
  const { address, phone, email, telegram, facebook, instagram } = props;
  const context = useContext(AuthContext);
  return (
    <div className="ContactsInfo">
      <h3>{labels[context.state.lang as string].pages.contacts.info}</h3>
      <hr />
      <div className="contact-option">
        <div className="icon-wrapper">
          <img
            src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Fcontact-page%2Faddress.png?alt=media&token=2fb28441-7270-4ebe-93eb-54bf0c7f1dfa"
            alt="address"
          />
        </div>{' '}
        {address}
      </div>
      <div className="contact-option">
        <div className="icon-wrapper">
          <img
            src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Fcontact-page%2Fmobile-phone.png?alt=media&token=a184bbb9-5b5b-47a9-bb33-d61740e95b9c"
            alt="phone"
          />
        </div>{' '}
        {phone}
      </div>
      {email !== undefined ? (
        <div className="contact-option">
          <div className="icon-wrapper">
            <img
              src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Fcontact-page%2Femail.png?alt=media&token=0e246b98-8137-4068-9594-d9441b571a06"
              alt="email"
            />
          </div>{' '}
          {email}
        </div>
      ) : null}
      {facebook !== undefined ||
      instagram !== undefined ||
      telegram !== undefined ? (
        <div className="socials">
          {' '}
          <hr />
          {facebook !== undefined ? (
            <div className="contact-option">
              <div className="icon-wrapper">
                <a href={facebook}>
                  <img
                    src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Fcontact-page%2Ffacebook.png?alt=media&token=5fdb89e0-3382-43af-af56-379ed4a5db2f"
                    alt="facebook"
                  />
                </a>
              </div>
            </div>
          ) : null}
          {instagram !== undefined ? (
            <div className="contact-option">
              <div className="icon-wrapper">
                <a href={instagram}>
                  <img
                    src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Fcontact-page%2Finstagram.png?alt=media&token=58c1731f-94f2-4d23-82e8-1e2d753413b7"
                    alt="instagram"
                  />
                </a>
              </div>
            </div>
          ) : null}
          {telegram !== undefined ? (
            <div className="contact-option">
              <div className="icon-wrapper">
                <a href={telegram}>
                  <img
                    src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Fcontact-page%2Ftelegram.png?alt=media&token=95badfef-90f7-49b8-858e-a92a03657f4e"
                    alt="telegram"
                  />
                </a>
              </div>{' '}
            </div>
          ) : null}
        </div>
      ) : null}
    </div>
  );
};

export default contactInfo;
