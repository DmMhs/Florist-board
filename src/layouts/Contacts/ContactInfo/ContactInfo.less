.ContactsInfo {
  width: 50%;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 8px;
  border: 1px solid #eee;
  box-shadow: 1px 1px 5px #ccc;
  color: #333333;
  h3 {
    font-weight: normal;
    font-size: 2rem;
    width: 75%;
  }
  hr {
    border: 0.5px solid #eee;
    width: 100%;
  }
  &:last-child {
    margin-bottom: 2rem;
  }
  .contact-option {
    display: flex;
    flex-direction: row;
    align-items: center;
    width: 90%;
    padding: 5px;
    font-size: 1.1rem;
    .icon-wrapper {
      width: 10%;
      margin-right: 1rem;
      img {
        margin-right: auto;
        max-width: 50%;
      }
    }
  }
  .socials {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;
    hr {
      width: 100%;
    }
    .contact-option {
      width: calc(90% / 3);
      box-sizing: border-box;
      .icon-wrapper {
        width: 100%;
        margin: 0;
        img {
          max-width: 15%;
        }
      }
    }
  }
}
@media screen and (max-width: 1050px) {
  .ContactsInfo {
    width: 700px;
  }
}
@media screen and (max-width: 768px) {
  .ContactsInfo {
    width: 420px;
    margin-bottom: 2rem;
    hr {
      margin: 0 auto;
    }
    .contact-option {
      font-size: 1rem;
      .icon-wrapper {
        width: 20%;
        img {
          max-width: 60%;
        }
      }
    }
    .socials {
      display: flex;
      flex-direction: row;
      justify-content: center;
      align-items: center;
      flex-wrap: wrap;
      hr {
        width: 100%;
      }
      .contact-option {
        width: calc(90% / 3);
        box-sizing: border-box;
        .icon-wrapper {
          width: 100%;
          margin: 0;
          img {
            max-width: 40%;
          }
        }
      }
    }
  }
}
@media screen and (max-width: 450px) {
  .ContactsInfo {
    width: 280px;
  }
}
