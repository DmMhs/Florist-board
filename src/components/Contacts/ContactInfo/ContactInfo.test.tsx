import React from 'react';
import { shallow, mount } from 'enzyme';

import ContactInfo from './ContactInfo';
import { BrowserRouter } from 'react-router-dom';
import AppContextProvider from '../../../AppContext';
import { notDeepEqual } from 'assert';

it('ContactInfo component matches a snapshot', () => {
  const wrapper = mount(
    <BrowserRouter>
      <AppContextProvider>
        <ContactInfo address="asd" phone="asfas" />
      </AppContextProvider>
    </BrowserRouter>
  );
  const context = wrapper.find('AppContextProvider').instance();
  context.setState({
    lang: 'en'
  });
  expect(wrapper).toMatchSnapshot();
});

it('Shows content conditionally', () => {
  const wrapper = mount(
    <BrowserRouter>
      <AppContextProvider>
        <ContactInfo address="asd" phone="asfas" instagram="asfaszvcn" />
      </AppContextProvider>
    </BrowserRouter>
  );
  const context = wrapper.find('AppContextProvider').instance();
  context.setState({
    lang: 'en'
  });
  expect(wrapper.find('.address-icon').exists()).toBeTruthy();
  expect(wrapper.find('.phone-icon').exists()).toBeTruthy();
  expect(wrapper.find('.facebook-icon').exists()).toBeFalsy();
  expect(wrapper.find('.instagram-icon').exists()).toBeTruthy();
  expect(wrapper.find('.telegram-icon').exists()).toBeFalsy();
});
