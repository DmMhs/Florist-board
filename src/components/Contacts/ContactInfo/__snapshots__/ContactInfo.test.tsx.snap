// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsInfo works as expected ContactInfo component matches a snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(AppContextProvider)>
      <Route>
        <AppContextProvider
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <contactInfo
            address="asd"
            phone="asfas"
          >
            <div
              className="ContactsInfo"
            >
              <h3>
                Feel free to get in touch with us!
              </h3>
              <hr />
              <div
                className="contact-option"
              >
                <div
                  className="icon-wrapper"
                >
                  <img
                    alt="address"
                    className="address-icon"
                    src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Fcontact-page%2Faddress.png?alt=media&token=2fb28441-7270-4ebe-93eb-54bf0c7f1dfa"
                  />
                </div>
                 
                asd
              </div>
              <div
                className="contact-option"
              >
                <div
                  className="icon-wrapper"
                >
                  <img
                    alt="phone"
                    className="phone-icon"
                    src="https://firebasestorage.googleapis.com/v0/b/florist-cb933.appspot.com/o/icons%2Fcontact-page%2Fmobile-phone.png?alt=media&token=a184bbb9-5b5b-47a9-bb33-d61740e95b9c"
                  />
                </div>
                 
                asfas
              </div>
            </div>
          </contactInfo>
        </AppContextProvider>
      </Route>
    </withRouter(AppContextProvider)>
  </Router>
</BrowserRouter>
`;
